content/docs/user-guide/testing/getting-started/_index.md

```diff
@@ -0,0 +1,232 @@
+            PRIVATE
+                AZ::AzTest
+                Legacy::CryCommon
+                Gem::MyModuleName
+        COMPONENT
+            MyModuleComponent
+    )
+
+    # Register new test module from Step 2
+    ly_add_googletest(
+        NAME MyModuleName.Tests
+    )
+
+endif()
+```
+
+#### Step 0: Add a production build target
+
+Before configuring tests, you should first define the library which gets tested. This is defined in a CMakeLists.txt, which typically lives in the same directory as the code or in a parent directory. For more information on configuring this, please read more about adding library targets at [Getting Started with the CMake Build System](/docs/user-guide/build/). Note that you only need to define a library that will be built, and can start configuring and writing tests before you finish writing the production code.
+
+#### Step 1: Add a test build target
+
+Similar to the production target, the test target defines a library in a CMakeLists.txt configuration file. Start by finding the CMakeLists.txt for your module, which should already be created from Step 0. It should exist at a path similar to `o3de/.../<MyModule>/CMakeLists.txt`
+
+Modify the CMakeLists.txt file to define your new test module, using the helper function `ly_add_target()`. Similar to the production build target, it is easiest to create another file which lists the C++ files used to compile the test library. The example above uses `dev/.../<MyModule>/mymodule_test_files.cmake`, which would have content similar to:
+
+```
+set(FILES
+    test/MyModuleTestFile.cpp
+    test/MyModuleMathTests.cpp
+)
+```
+
+#### Step 2: Register the test module
+
+The final CMakeLists.txt step is to register the module with CTest. For this, use helper function `ly_add_googletest()`
+
+To verify everything is set up correctly, run the [CMake configure command](/docs/user-guide/build/configure-and-build/) from the CMake CLI or CMake GUI. This will register everything you just added, and emit errors if anything was misconfigured.
+
+#### Step 3: Write new tests
+
+Now that you have configured CMake to create a library from your tests, and registered this library with CTest, you are ready to write new tests. To simplify your module structure, create new test files inside `o3de/.../<MyModule>/tests/`
+
+Tests are written using standard [GoogleTest](https://github.com/google/googletest/blob/main/docs/index.md) syntax, which helps you write small functions to test your code. To pull in everything from GoogleTest plus a few convienient tools, add the following statement to your file:
+
+```cpp
+#include <AzTest/AzTest.h>
+```
+
+To keep test functions legible at a glance, we recommend using the [Osherove Naming Convention](https://osherove.com/blog/2005/4/3/naming-standards-for-unit-tests.html) of `UnitOfWork_StateUnderTest_ExpectedBehavior`. This can be especially helpful when reading a report with many individual test case failures. In a short a C++ example:
```

Comment:
Need to resolve a minor conflict between this naming recommendation, and one from GoogleTest: http://google.github.io/googletest/faq.html#why-should-test-suite-names-and-test-names-not-contain-underscore

Reply:
Can we also mention that this is only a recommendation and that not all tests follow this approach. A lot of the tests I've written use a style where we just describe the behavior under test (e.g. `MouseMoveCorrectlyTransformsCursorPositionInGlobalAndLocalSpace`) I don't think we should encourage one over the other necessarily, it can also be useful to potentially make the sentence more readable to allow this - `Mouse_move_correctly_transforms_cursor_position_in_global_and_local_space` (this doesn't have to explicitly be in the docs, but I think we should have more than example of test names.

Reply:
(loads of good advice on this in this book - https://www.amazon.co.uk/Modern-Programming-Test-Driven-Development-Better/dp/1937785483/ref=sr_1_1?crid=39B88RBW10B11&keywords=Modern+C%2B%2B+Programming+with+Test-Driven+Development%3A+Code+Better%2C+Sleep+Better&qid=1641490223&sprefix=modern+c%2B%2B+programming+with+test-driven+development+code+better+sleep+better%2Caps%2C398&sr=8-1) 🙂

Reply:
I also like this general advice

> A test should describe a behaviour you would like to test
> The test name should form a sentence describing the behaviour under test
> Avoid using words like 'Check', 'Verify', 'Assert' - they are redundant and clutter the test name

Reply:
+1 on clarifying the recommendation. Though I don't want to go too deep on general advice in the getting-started guide, as there is another best-practices guide planned and that information is slightly more of an  advanced concept.

...perhaps naming advice should not even be mentioned here?

Reply:
Got you, yes this can totally be in the more advanced guide that would be great to link to from here. I agree maybe talking about naming at all can be reserved for that doc?